scripts/release/other_identifiers/indirectIdentifiers_from_mart.pl

```diff
@@ -81,51 +81,51 @@
 
 foreach my $rpg_id (keys %accs) {
     foreach my $i (@{$accs{uc($rpg_id)}}) {
-	eval {
-	    my $ref_db = $i->referenceDatabase->[0]->name->[0];
-	    
-	    no warnings 'uninitialized';
-	    my @ensp;
-	    if ($ref_db =~ /ensembl/i) {
-		push @ensp, $rpg_id;
-	    } elsif ($ref_db =~ /uniprot/i) {
-		@ensp = @{$uniprot_to_ensp{$rpg_id}};
-	    }
-	
-	    foreach my $ensp (@ensp) {
-		my $enst = $ensp_to_enst{$ensp};
-		my $ids = $enst_to_ids{$enst};
-		delete $ids->{$rpg_id};
-    
-		$i->add_attribute_value_if_necessary('otherIdentifier', $enst, keys %$ids);
-	    }
-	};
+        eval {
+            my $ref_db = $i->referenceDatabase->[0]->name->[0];
+
+            no warnings 'uninitialized';
+            my @ensp;
+            if ($ref_db =~ /ensembl/i) {
+                push @ensp, $rpg_id;
+            } elsif ($ref_db =~ /uniprot/i) {
+                @ensp = @{$uniprot_to_ensp{$rpg_id}};
+            }
+
+            foreach my $ensp (@ensp) {
+                my $enst = $ensp_to_enst{$ensp};
+                my $ids = $enst_to_ids{$enst};
+                delete $ids->{$rpg_id};
+
+                $i->add_attribute_value_if_necessary('otherIdentifier', $enst, keys %$ids);
+            }
+        };
     }
 }
 
 while (my ($acc,$ar) = each %accs) {
     foreach my $i (@{$ar}) {
-	my @otherIdentifiers = sort {fc($a) cmp fc($b)} grep {defined} @{$i->OtherIdentifier};
-	$i->OtherIdentifier(undef);
-	$i->OtherIdentifier(@otherIdentifiers);
-	
-	print $acc, "\t", join(',',@otherIdentifiers), "\n";
-	$dba->update_attribute($i,'otherIdentifier');
+        my @otherIdentifiers = sort {fc($a) cmp fc($b)} grep {defined} @{$i->OtherIdentifier};
+        $i->OtherIdentifier(undef);
+        $i->OtherIdentifier(@otherIdentifiers);
+
+        print $acc, "\t", join(',',@otherIdentifiers), "\n";
+        $dba->update_attribute($i,'otherIdentifier');
     }
 }
 
 print "$0: no fatal errors.\n";
 
 sub get_prefix {
     my $resource = shift;
-    
+
     if ($resource =~ /_locuslink/i) {
-	return 'LocusLink';
+        return 'LocusLink';
     } elsif ($resource =~ /_entrezgene/i) {
-	return 'EntrezGene:';
+        return 'EntrezGene:';
     } elsif ($resource =~ /_mim/i) {
-	return 'MIM:';
+        return 'MIM:';
     }
```

Comment:
Should there be a final 'else' that says unknown prefix was found?

Reply:
It does return an empty string if none of the conditions match.  No prefix found is expected for most identifiers in which case a prefix isn't needed.